File: 2825-make-string-a-subsequence-using-cyclic-increments/2825-make-string-a-subsequence-using-cyclic-increments.cpp
class Solution {
public:
    bool canMakeSubsequence(string str1, string str2) {
        int it1=0,it2=0;
        while(it1<str1.size()&&it2<str2.size())
        {
            if(str1[it1]==str2[it2])
            {
                it1++;
                it2++;
                continue;
            }
            char ch=str1[it1];
            if(ch=='z')
                ch='a';
            else 
                ch++;
            if(str2[it2]==ch)
            {
                it1++;
                it2++;
                continue;
            }
            it1++;
        }
        return it2==str2.size();
    }
};

File: 2825-make-string-a-subsequence-using-cyclic-increments/2825-make-string-a-subsequence-using-cyclic-increments.java
class Solution {
    public boolean canMakeSubsequence(String str1, String str2) {
        int count=0;
        for(int i=0;i<str1.length();i++)
        {
            int s1=str1.charAt(i)-'a';
            int s2=str2.charAt(count)-'a';
            if((s1+1)%26==s2)
            {
                count++;
            }
            else if(count<str2.length()&&str1.charAt(i)==str2.charAt(count))
            {
                count++;
            }
            if(count==str2.length())
                    return true;
        }
        return false;
    }
}